src/corelibs/U2View/src/ov_msa/align_to_alignment/AlignSequencesToAlignmentSupport.cpp

```diff
@@ -97,7 +97,9 @@ void AlignSequencesToAlignmentAction::sl_activate() {
             AppContext::getTaskScheduler()->registerTopLevelTask(task);
         }
     } else {
-        QString filter = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::SEQUENCE, true);
+        QString f1 = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::MULTIPLE_SEQUENCE_ALIGNMENT, false);
+        QString f2 = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::SEQUENCE, true);
```

Comment:
Is the result list sorted or not? May be pass both types (list of types) to the prepareDocumentsFileFilterByObjType ?

Reply:
> Is the result list sorted or not?

The results are not sorted, they appeared in the combo box in the order they have been added

> May be pass both types (list of types) to the prepareDocumentsFileFilterByObjType ?

Sorry, don't gent the idea. You mean, to pass the argument like this?

`DialogUtils::prepareDocumentsFileFilterByObjType( { GObjectTypes::MULTIPLE_SEQUENCE_ALIGNMENT, GObjectTypes::SEQUENCE }, true);`

Reply:
Yes, so the filters are sorted and the (*.* All) is the first.

Reply:
After this change we have 2 FASTA items in the file filter lis
![fasta-2x](https://user-images.githubusercontent.com/17791039/134462956-c7351314-f800-42a8-9353-31d790478056.jpg)
t

Reply:
Hm, yes, this problem really exists. `QFileDialog` doesn't remove duplicate filters on it's own, unfortunately. I can suggest you 2 solutions:
1. The local one. I will remove duplicates from every place in this commit, where I have added a double-filters like this:
    ```
    QString f1 = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::MULTIPLE_SEQUENCE_ALIGNMENT, false);
    QString f2 = DialogUtils::prepareDocumentsFileFilterByObjType(GObjectTypes::SEQUENCE, true);
    QString filter = f2 + "\n" + f1;
```

2. The global one. I will add a new function `QString removeDuplicatedFilters(const QString &oldFilter)` to `U2FileDialog` and call this function in `U2FileDialog::getOpenFileName`, `U2FileDialog::getOpenFileNames` and `U2FileDialog::getSaveFileName` before `QFileDialog` creates the dialog.

Which one do I need to use?

Reply:
This is what I suggested in the first comment: a single function that accepts a list of object types.